### ISSUE TITLE
Property 'children' does not exist on type

### ISSUE THREAD

**Issue by jroitgrund** [palantir/tslint#1780]
### Bug Report

- __TSLint version__: `3.15.1`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: CLI / Visual Studio

#### TypeScript code being linted

```ts
import { assert } from "chai";
import * as React from "react";
import * as TestUtils from "react-addons-test-utils";
import * as sinon from "sinon";

import { Register } from "../../../src/register/register";

describe("Register", () => {
    let register: React.ReactElement<Register>;
    let fetchUserSecret: sinon.SinonSpy;
    const location = { query: { token: "myToken" } };
    const user = {
        secret: "secret",
        userId: "userId",
    };
    let renderer: React.ShallowRenderer;

    beforeEach(() => {
        renderer = TestUtils.createRenderer();
        fetchUserSecret = sinon.spy();
        renderer.render(
            <Register
                fetchUserSecret={fetchUserSecret}
                user={user}
                location={location}
                qrCodeImgTag="<img class='img-class' name='foo'>"
            />
        );
        register = renderer.getRenderOutput();
    });

    it("calls fetchUserSecret", () => {
        sinon.assert.calledWith(fetchUserSecret, "myToken");
    });

    it("renders the passed HTML", () => {
        assert.equal(register.props.children[0].props.name, "foo");
        assert.equal(register.props.children[1].props.children[1], "userId");
        assert.equal(register.props.children[2].props.children[1], "secret");
    });
});
```

with `tslint.json` configuration:

```json
{
  "extends": [
    "tslint:latest",
    "tslint-react"
  ],
  "rules": {
    "jsx-alignment": true,
    "jsx-no-lambda": true,
    "jsx-no-string-ref": true
  }
}
```

#### Actual behavior

I get an error for the three `register.props.children` lines: Property `children` does not exist on type `Register`.

#### Expected behavior

I don't get this error because

a) Register is a react component, so it has that property

b) The test compiles and passes !

**Comment by adidahiya** [comment]
That looks like a compiler error, not a TSLint failure. What is the output of the CLI?

**Comment by adidahiya** [comment]
resolved offline, it is a compiler error
